### PR TITLE
Profile Consolidation

### PR DESCRIPTION
**Pull request by jasonfrancis**
Consolidating the Profile pages to use methods in the `PageBase` class and to use the `SuggestionButtons` component.

Modified the `PageBase` GET/POST/PUT/DELETE methods to also return the raw response so we can make decisions based on the actual response in our child pages.

Fixed a bug in `PageBase` where if we got an error from the server we wouldn't return the response's actual `EntityPermissions`.

### CODE REVIEW

File: src/Web/Pages/Profile/EditInterests.razor [hunk 1]
```diff
@@ -171,70 +158,41 @@
...
-			var url = $"/people/{UpdatedPerson.Id}";
-
-			using (var apiClient = Api())
+			using(Loading())
 			{
-				var reqJson = JsonConvert.SerializeObject(objToPut, Json.JsonSerializerSettings);
-				var reqContent = new StringContent(reqJson, System.Text.Encoding.UTF8, "application/json");
-				var response = await apiClient.PutAsync(url, reqContent);
+				// Make an in-memory copy of Person to PUT to the server.
+				var objToPut = JsonConvert.DeserializeObject<Person>(JsonConvert.SerializeObject(UpdatedPerson, Json.JsonSerializerSettings), Json.JsonSerializerSettings);
```
Comment (jardrake) [inline src/Web/Pages/Profile/EditInterests.razor:174]: I don't think this is the first time we've had to pull this off. We might be able to put this up into pageBase or something

File: src/Web/Pages/Profile/PersonProfile.razor [hunk 2]
```diff
@@ -149,57 +123,49 @@
...
+		var matchedMembership = Memberships.FirstOrDefault(m => m.UnitId == unit.Id);
+		
+		if(matchedMembership != null)
 		{
-			GeneralException = new Exception($"Failed to fetch UnitMemberships for {NetId}.");
-			return;
+			output += "<div>\n";
+			output += $"\t<strong>Title:</strong> {matchedMembership.Title}\n";
+			output += "</div>\n";
+			output += "<div>\n";
```
Comment (jardrake) [inline src/Web/Pages/Profile/PersonProfile.razor:163]: A component would be best here

File: src/Web/Pages/Profile/EditInterests.razor (hunk 1 above)
Comment (jasonfrancis) [reply src/Web/Pages/Profile/EditInterests.razor:174]: We don't need to do it all the time, but there are cases where we don't want to fiddle withe the original object.  Maybe add a `PageBase.MakeCopy(Entity source)` method?

File: src/Web/Pages/Profile/PersonProfile.razor (hunk 2 above)
Comment (jasonfrancis) [reply src/Web/Pages/Profile/PersonProfile.razor:163]: I'll whip one up.

File: src/Web/Pages/Profile/EditInterests.razor (hunk 1 above)
Comment (jardrake) [reply src/Web/Pages/Profile/EditInterests.razor]: Added DeepCopy to PageBase. Resolved

File: src/Web/Pages/Profile/PersonProfile.razor (hunk 2 above)
Comment (jasonfrancis) [reply src/Web/Pages/Profile/PersonProfile.razor]: Fixed with commit 074b3f1

File: src/Web/Pages/Profile/EditInterests.razor (hunk 1 above)
Comment (jasonfrancis) [reply src/Web/Pages/Profile/EditInterests.razor]: Implemented in 2ec6572 and 25866fa

**Pull request merged**